tcl/pd_widget.tcl

```diff
@@ -80,9 +80,9 @@ proc ::pd::widget::_call {behaviour obj args} {
 }
 
 #
-proc ::pd::widget::create {type obj cnv} {
+proc ::pd::widget::create {type obj cnv posX posY} {
```

Comment:
What's the use of `posX` `posY` if the width/height are unspecified? See `bang::create()` : you are left with a zero-width/height element placed somewhere else than `(0, 0)`, but that's not much more useful than having it at `(0, 0)`... If `::create()` just becomes a shorthand for also initializing an object's properties, then why not giving it full opt parsing capabilities like `config`? Or take `args` that can be type-specific?

Reply:
to quote from the commit message:
> add initial position to ::pd::widget::create
>
>    - this matches the object-creation on-disk more closely ("#X obj 100 100 foo ...")
 >   - we need to set an initial position anyhow...
>    - saves us a line on the core :-)

my vague idea is:
- Pd itself knows where the object is to be positioned, for all types of "widgets".
  i'd like to think that this constructor could be called by Pd, rather than the object's "widgetbehaviour".
  (in reality this is of course not true; for Pd a `[bng]` object looks very much like a `[pack]` object - both are of type `obj`)
- all the other properties seem to be really objectclass-specific, so it makes sense to keep these two apart
- the (GUI)constructor is typically called only "once", whereas the properties might change more often. rather than duplicating the arguments in the calls to both the `create` and the `config` proc (from a Pd-core POV), it seemed to me nicer to just call the `config` in addition to `create`.
- i also have this vague idea that a single GUI-widget can live on multiple canvases at the same time. while their config (and relative movement) can be shared across the canvases (i think), this is not true for their initial position (see GOP: on the parent canvas, your numberbox might live on +100+100, whereas on the subpatch it might live on +200+532)